src/DruidDataSource.ts

```diff
@@ -1,34 +1,329 @@
+       * SELECT biz FROM boz
+       *
+       * Ex UNION-ALL result:
+       * SELECT foo FROM bar
+       * UNION ALL
+       * SELECT biz FROM boz
+       */
+      throw new Error('Multi-query not yet supported');
+    }
+
+    const initialWhere = (() => {
+      if (query.where) {
+        return query.where;
+      }
+      const firstAdhoc = adhocFilters.pop();
+      return {
+        type: 'binary_expr',
+        operator: firstAdhoc?.operator,
+        left: {
+          type: 'column_ref',
+          table: null,
+          column: firstAdhoc?.key,
+        },
+        right: {
+          type: 'single_quote_string',
+          value: firstAdhoc?.value,
+        },
+      } as SqlExpression;
+    })();
+
+    query.where = adhocFilters.reduce<SqlExpression>((acc, filter) => {
+      const expression = ((): SqlExpression => {
+        if (filter.operator === '<' || filter.operator === '>') {
+          return {
+            type: 'binary_expr',
+            operator: filter.operator,
+            left: {
+              type: 'column_ref',
+              table: null,
+              column: filter.key,
+            },
+            right:
+              typeof filter.value === 'number'
+                ? {
+                    type: 'number',
+                    value: filter.value,
+                  }
+                : {
+                    type: 'single_quote_string',
+                    value: String(filter.value),
+                  },
+          };
+        } else if (filter.operator === '!~') {
+          return {
+            type: 'binary_expr',
+            operator: 'NOT LIKE',
+            left: {
+              type: 'column_ref',
+              table: null,
+              column: filter.key,
+            },
+            right: {
+              type: 'single_quote_string',
+              value: `%${filter.value}%`,
+            },
+          };
+        } else if (filter.operator === '=~') {
+          return {
+            type: 'binary_expr',
+            operator: 'LIKE',
+            left: {
+              type: 'column_ref',
+              table: null,
+              column: filter.key,
+            },
+            right: {
+              type: 'single_quote_string',
+              value: `%${filter.value}%`,
+            },
+          };
         }
-        return match;
+        return {
+          type: 'binary_expr',
+          operator: filter.operator,
+          left: {
+            type: 'column_ref',
+            table: null,
+            column: filter.key,
+          },
+          right: {
+            type: 'single_quote_string',
+            value: String(filter.value),
+          },
+        };
+      })();
+
+      return {
+        type: 'binary_expr',
+        operator: 'AND',
+        left: acc,
+        right: expression,
+      };
+    }, initialWhere);
+
+    return {
+      ...templatedQuery,
+      builder: {
+        ...templatedQuery.builder,
+        query: SQL.stringify(query),
```

Comment:
Do we know that this stringification is safe and can't lead to possible SQL injection attacks?